File: hard/string-searching/string.pl
open(INFILE, "<$ARGV[0]") or die("Cannot open file $ARGV[0] for reading: $!");
while(chomp(my $line = <INFILE>)) {
    next if($line =~ m/^s$/); # skip blank lines
    ($word, $search) = split(',', $line);
    unless($search =~ m/\\\*/) {
        $search =~ s/\*/.*/;
    }
    
    if($word =~ m/$search/) {
        print "true\n";    
    }
    else {
        print "false\n";
    }
}
close(INFILE);

File: hard/string-searching/string.py
import sys, re
test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    word, search = test.strip().split(',')
    search = search.replace('*', '[\w\W]*')
    print 'true' if re.search(search, word) else 'false'
test_cases.close()
